packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx

```diff
@@ -459,7 +483,9 @@ export const EntityHeaderTable = (props: EntityHeaderTableProps) => {
         confirmButtonCopy={`Add ${pluralObjectName}`}
         onConfirm={items => {
           if (hideTextFieldToPasteValue) {
-            onUpdate([...refsInState, ...items])
+            const newRefs = [...refsInState, ...items]
+            setRefsInState(newRefs)
+            onUpdate(newRefs)
```

Comment:
We have a weird local/parent copied state issue here, so make this small change to try to keep them in sync